### DISCUSSION TITLE
【请问】启动APP时，可以默认附屏吗？

### DISCUSSION THREAD

**Discussion by kakapas** [ClassicOldSong/Apollo#154] [General]
1、app中不管选desktop还是virtual desktop，都会自动切换成app作为主屏。即使我在电脑端把电脑屏幕设置为主显示，退出app后再次进入后，又把app作为主显示了。不能默认设置app端为附屏吗？
2、app端控制电脑端的屏幕，点击下，就点不动，没法用app端控制电脑吗？
谢谢~

**Comment by ClassicOldSong** [comment]
1. 请参考 [FAQ](https://github.com/ClassicOldSong/Apollo#faq) 中的 `Primary display changed to the virtual display after connection. I don't want that.` 条目。
2. 请在 PIN 页面中给予对应客户端鼠标/键盘输入权限。

**Reply by kakapas** [reply]
全部权限都授予了，点击下就没反应了。。。手机，平板都这样，不知道咋回事。。。

**Reply by ClassicOldSong** [reply]
什么手机，用的什么客户端，PC系统版本/硬件信息/装了什么杀毒软件/游戏反作弊系统？

启动Apollo的时候请不要直接双击.exe，请务必通过安装时创建的快捷方式启动，否则会有权限问题，鼠标点到管理员权限的窗口里以后会无法控制。

**Reply by kakapas** [reply]
我刚才在家里和单位的2个电脑上重新试了下
1、手机红米k70，平板联想小新Pad Pro，app都用的Artemis
2、2个电脑都是win10 22H2(操作系统内部版本 19045.3208)
3、安装时候都完全关闭了杀毒软件，没有反作弊
4、安装最新的Apollo v0.2.6-alpha.3的时候，会提示卡在这里：https://github.com/nefarius/ViGEmBus/releases/download/v1.22.0/ViGEmBus_1.22.0_x64_x86_arm64.exe

所以我已经先装了ViGEmBus_1.22.0_x64_x86_arm64.exe
但还是会卡在这里（我看了，这个文件是存在的：C:\Windows\System32\drivers\ViGEmBus.sys）


安装时候的提示如下：

输出目录: C:\Program Files\Apollo
解压缩: sunshine.exe... 100%
解压缩: zlib1.dll... 100%
输出目录: C:\Program Files\Apollo\tools
解压缩: ddprobe.exe... 100%
解压缩: sunshinesvc.exe... 100%
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo\assets
解压缩: apps.json... 100%
输出目录: C:\Program Files\Apollo\assets\shaders
输出目录: C:\Program Files\Apollo\assets\shaders\directx
解压缩: convert_yuv420_packed_uv_type0_ps.hlsl... 100%
解压缩: convert_yuv420_packed_uv_type0_ps_linear.hlsl... 100%
解压缩: convert_yuv420_packed_uv_type0_ps_perceptual_quantizer.hlsl... 100%
解压缩: convert_yuv420_packed_uv_type0_vs.hlsl... 100%
解压缩: convert_yuv420_packed_uv_type0s_ps.hlsl... 100%
解压缩: convert_yuv420_packed_uv_type0s_ps_linear.hlsl... 100%
解压缩: convert_yuv420_packed_uv_type0s_ps_perceptual_quantizer.hlsl... 100%
解压缩: convert_yuv420_packed_uv_type0s_vs.hlsl... 100%
解压缩: convert_yuv420_planar_y_ps.hlsl... 100%
解压缩: convert_yuv420_planar_y_ps_linear.hlsl... 100%
解压缩: convert_yuv420_planar_y_ps_perceptual_quantizer.hlsl... 100%
解压缩: convert_yuv420_planar_y_vs.hlsl... 100%
解压缩: convert_yuv444_packed_ayuv_ps.hlsl... 100%
解压缩: convert_yuv444_packed_ayuv_ps_linear.hlsl... 100%
解压缩: convert_yuv444_packed_vs.hlsl... 100%
解压缩: convert_yuv444_packed_y410_ps.hlsl... 100%
解压缩: convert_yuv444_packed_y410_ps_linear.hlsl... 100%
解压缩: convert_yuv444_packed_y410_ps_perceptual_quantizer.hlsl... 100%
解压缩: convert_yuv444_planar_ps.hlsl... 100%
解压缩: convert_yuv444_planar_ps_linear.hlsl... 100%
解压缩: convert_yuv444_planar_ps_perceptual_quantizer.hlsl... 100%
解压缩: convert_yuv444_planar_vs.hlsl... 100%
解压缩: cursor_ps.hlsl... 100%
解压缩: cursor_ps_normalize_white.hlsl... 100%
解压缩: cursor_vs.hlsl... 100%
输出目录: C:\Program Files\Apollo\assets\shaders\directx\include
解压缩: base_vs.hlsl... 100%
解压缩: base_vs_types.hlsl... 100%
解压缩: common.hlsl... 100%
解压缩: convert_base.hlsl
解压缩: convert_linear_base.hlsl... 100%
解压缩: convert_perceptual_quantizer_base.hlsl
解压缩: convert_yuv420_packed_uv_ps_base.hlsl... 100%
解压缩: convert_yuv420_planar_y_ps_base.hlsl... 100%
解压缩: convert_yuv444_ps_base.hlsl... 100%
输出目录: C:\Program Files\Apollo\scripts
解压缩: install-service.bat... 100%
解压缩: migrate-config.bat... 100%
解压缩: uninstall-service.bat... 100%
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo\scripts
解压缩: autostart-service.bat
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo\drivers
输出目录: C:\Program Files\Apollo\drivers\sudovda
解压缩: SudoVDA.dll... 100%
解压缩: SudoVDA.inf... 100%
解压缩: code_signing.crt... 100%
解压缩: install.bat... 100%
解压缩: nefconc.exe... 100%
解压缩: sudovda.cat... 100%
解压缩: uninstall.bat... 100%
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo\scripts
解压缩: add-firewall-rule.bat... 100%
解压缩: delete-firewall-rule.bat... 100%
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo\scripts
解压缩: install-gamepad.bat... 100%
解压缩: uninstall-gamepad.bat... 100%
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo\tools
解压缩: audio-info.exe... 100%
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo\tools
解压缩: dxgi-info.exe... 100%
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo\assets
解压缩: box.png... 100%
解压缩: desktop-alt.png... 100%
解压缩: desktop.png... 100%
解压缩: steam.png... 100%
解压缩: virtual_desktop.png
输出目录: C:\Program Files\Apollo\assets\web
解压缩: apps.html... 100%
解压缩: config.html... 100%
解压缩: index.html... 100%
解压缩: login.html... 100%
解压缩: password.html... 100%
解压缩: pin.html... 100%
解压缩: troubleshooting.html... 100%
解压缩: welcome.html... 100%
输出目录: C:\Program Files\Apollo\assets\web\assets
解压缩: Navbar-13079613.css... 100%
解压缩: Navbar-96eaa329.js... 100%
解压缩: PlatformLayout-c2469f32.js... 100%
解压缩: ResourceCard-e3a0bd15.js... 100%
解压缩: _plugin-vue_export-helper-c27b6911.js
解压缩: apps-65dc2a61.js... 100%
解压缩: config-9376e968.js... 100%
解压缩: fa-brands-400-bc844b5b.ttf... 100%
解压缩: fa-brands-400-c411f119.woff2
解压缩: fa-regular-400-64f9fb62.ttf... 100%
解压缩: fa-regular-400-c732f106.woff2
解压缩: fa-solid-900-1f0189e0.woff2
解压缩: fa-solid-900-31f099c1.ttf... 100%
解压缩: fa-v4compatibility-2aca24b3.woff2
解压缩: fa-v4compatibility-a6274a12.ttf... 100%
解压缩: index-437b90a8.css
解压缩: index-91ea122c.js... 100%
解压缩: init-33bef63e.js... 100%
解压缩: init-e3be04f7.css... 100%
解压缩: login-4b0a66d5.js... 100%
解压缩: password-71e9726c.js... 100%
解压缩: pin-53f1abf5.js... 100%
解压缩: troubleshooting-1c7441ee.js... 100%
解压缩: welcome-1cbe98f4.js... 100%
输出目录: C:\Program Files\Apollo\assets\web\assets\css
解压缩: apollo.css... 100%
输出目录: C:\Program Files\Apollo\assets\web\assets\js
解压缩: qrcode.min.js... 100%
输出目录: C:\Program Files\Apollo\assets\web\assets\locale
解压缩: de.json... 100%
解压缩: en.json... 100%
解压缩: en_GB.json... 100%
解压缩: en_US.json... 100%
解压缩: es.json... 100%
解压缩: fr.json... 100%
解压缩: it.json... 100%
解压缩: ja.json... 100%
解压缩: pt.json... 100%
解压缩: ru.json... 100%
解压缩: sv.json... 100%
解压缩: tr.json... 100%
解压缩: zh.json... 100%
输出目录: C:\Program Files\Apollo\assets\web\images
解压缩: apollo-locked-16.png
解压缩: apollo-locked-45.png... 100%
解压缩: apollo-locked.ico... 100%
解压缩: apollo-locked.png... 100%
解压缩: apollo-locked.svg... 100%
解压缩: apollo-pausing-16.png
解压缩: apollo-pausing-45.png... 100%
解压缩: apollo-pausing.ico... 100%
解压缩: apollo-pausing.png... 100%
解压缩: apollo-pausing.svg... 100%
解压缩: apollo-playing-16.png
解压缩: apollo-playing-45.png
解压缩: apollo-playing.ico... 100%
解压缩: apollo-playing.png... 100%
解压缩: apollo-playing.svg... 100%
解压缩: apollo.ico... 100%
解压缩: logo-apollo-16.png
解压缩: logo-apollo-45.png... 100%
解压缩: logo-apollo.svg... 100%
输出目录: C:\Program Files\Apollo
输出目录: C:\Program Files\Apollo
创建卸载程序: C:\Program Files\Apollo\Uninstall.exe
Set install registry entry: 'DisplayName' to 'Apollo'
Set install registry entry: 'DisplayVersion' to '0.2.6-alpha.3'
Set install registry entry: 'Publisher' to 'SudoMaker'
Set install registry entry: 'UninstallString' to '"C:\Program Files\Apollo\Uninstall.exe"'
Set install registry entry: 'NoRepair' to '1'
Set install registry entry: 'NoModify' to '1'
Set install registry entry: 'DisplayIcon' to 'C:\Program Files\Apollo\sunshine.exe'
"创建文件夹: " C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Apollo
创建快捷方式: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Apollo\Apollo.lnk
创建快捷方式: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Apollo\Uninstall.lnk
Set install registry entry: 'StartMenu' to 'Apollo'
已处理的文件: C:\Program Files\Apollo
已成功处理 1 个文件; 处理 0 个文件时失败
================
Installing cert for the SudoVDA driver...
root "受信任的根证书颁发机构"
签名与公钥匹配
CertUtil: -addstore 失败: 0x80070005 (WIN32: 5 ERROR_ACCESS_DENIED)
CertUtil: 拒绝访问。
TrustedPublisher "受信任的发布者"
签名与公钥匹配
CertUtil: -addstore 失败: 0x80070005 (WIN32: 5 ERROR_ACCESS_DENIED)
CertUtil: 拒绝访问。
================
Removing the old driver... It's OK to show an error if you're installing the driver for the first time.
2024-12-16 19:46:10,546 ERROR [default] Failed to delete device node, error: 没有可用的数据了。

Failed to delete device node, error: LEER
================
Installing the new driver...
2024-12-16 19:46:10,718 INFO [default] Device node created successfully
Device node created successfully
2024-12-16 19:46:11,313 ERROR [default] Failed to install driver, error: LEER
Failed to install driver, error: LEER
================
Done!
已处理的文件: C:\Program Files\Apollo\config
已成功处理 1 个文件; 处理 0 个文件时失败
已处理的文件: C:\Program Files\Apollo\config\credentials
已成功处理 1 个文件; 处理 0 个文件时失败
已处理的文件: C:\Program Files\Apollo\config\credentials
已成功处理 1 个文件; 处理 0 个文件时失败
已处理的文件: C:\Program Files\Apollo\config\credentials
已成功处理 1 个文件; 处理 0 个文件时失败
找不到 C:\Program Files\Apollo\*.txt
找不到 C:\Program Files\Apollo\*.log
确定。

确定。

Get-Item : 找不到路径“C:\Windows\System32\drivers\ViGEmBus.sys”，因为该路径不存在。
所在位置 行:1 字符: 13
+ Exit $(if ((Get-Item $env:SystemRoot\System32\drivers\ViGEmBus.sys).V ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Windows\System32\drivers\ViGEmBus.sys:String) [Get-Item], ItemNotFou 
   ndException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand
 
https://github.com/nefarius/ViGEmBus/releases/download/v1.22.0/ViGEmBus_1.22.0_x64_x86_arm64.exe




5、不管错误提示，等一下，最后还是会安装完成
如果用开始菜单的快捷方式启动，会出现如下错误提示
然后无法启动
![err2](https://github.com/user-attachments/assets/12b50120-118c-47dc-9215-bbe15a65accb)



6、直接用exe启动后
用app控制电脑
我又试了下

进去后
有时候可以控制浏览器
但是只要切换到资源管理器就点不动了

有时候进去，完全点不动

摸不到规律

**Comment by ClassicOldSong** [comment]
你电脑的权限不太正常。是不是装的某电脑城快速装机版系统？建议使用官方镜像重装系统，以及建议升级Win11。

**Reply by kakapas** [reply]
我是原版系统安装的。

**Comment by kakapas** [comment]
搞定了，要右键选择管理员模式进行安装，就没错了。
另外请教一下，手机、平板不能同时连接吗？我试了下，只要一个连接上，另外一个就会被强制下线。

**Comment by kakapas** [comment]
搞定了，要右键选择管理员模式进行安装，就没错了。
另外请教一下，手机、平板不能同时连接吗？我试了下，只要一个连接上，另外一个就会被强制下线。

**Reply by ClassicOldSong** [reply]
没给管理员权限你是怎么装上的。。。。。。

手机/平板只能同时串流一个屏幕，如果想分别串流不同屏幕的话请阅读[WiKi](https://github.com/ClassicOldSong/Apollo/wiki/How-to-start-multiple-instances-of-Apollo)

**Reply by kakapas** [reply]
谢谢~
